Creating OCP performance guide (#564)
RedHatInsights/red-hat-ansible-automation-platform-documentation: downstream/modules/platform/con-alternative-capacity-limits.adoc

[id="con-alternative-capacity-limits"]

= Alternative capacity limiting with {ControllerName} settings

The capacity of a control node in Openshift is determined by the memory and CPU limits. 
However, if these are not set then the capacity is determined by the memory and CPU detected by the pod on the filesystem, which are actually the CPU and memory of the underlying kubernetes node.

This can lead to issues with overwhelming the underlying kubernetes pod if the {ControllerName} pod is not the only pod on that node. 

> If you do not want to set limits directly on the task container, you can provide the following by using  extra_settings, see _Extra Settings_ in xref:proc-set-custom-pod-timeout[Custom pod timeouts] section for how to configure the following

Reviewer: ```suggestion
If you do not want to set limits directly on the task container, you can use  extra_settings. Refer to  _Extra Settings_ in xref:proc-set-custom-pod-timeout[Custom pod timeouts] for how to configure the following:
```

[options="nowrap" subs="+quotes,attributes"]
----
SYSTEM_TASK_ABS_MEM
SYSTEM_TASK_ABS_CPU
----

> This can be thought of as a soft limit within the application that enables {ControllerName} to control how much work it attempts to run, while not risking any CPU throttling from Kubernetes itself, or being reaped if memory usage peaks above the desired limit. 

Reviewer: ```suggestion
This acts as a soft limit within the application that enables {ControllerName} to control how much work it attempts to run, while not risking any CPU throttling from Kubernetes itself, or being reaped if memory usage peaks above the desired limit. 
```
